media/rtmp2segment.go

```diff
@@ -18,30 +18,41 @@ import (
 	"time"
 
 	"github.com/livepeer/go-livepeer/clog"
+	"github.com/livepeer/go-livepeer/mediaserver"
 	"github.com/livepeer/lpms/ffmpeg"
 	"golang.org/x/sys/unix"
 )
 
 var waitTimeout = 20 * time.Second
 
 type MediaSegmenter struct {
-	Workdir string
+	Workdir        string
+	MediaMTXClient *mediaserver.MediaMTXClient
+	MediaMTXHost   string
 }
 
-func (ms *MediaSegmenter) RunSegmentation(in string, segmentHandler SegmentHandler) {
+func (ms *MediaSegmenter) RunSegmentation(ctx context.Context, in string, segmentHandler SegmentHandler, id, sourceType string) {
 	outFilePattern := filepath.Join(ms.Workdir, randomString()+"-%d.ts")
 	completionSignal := make(chan bool, 1)
 	wg := &sync.WaitGroup{}
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		processSegments(segmentHandler, outFilePattern, completionSignal)
+		processSegments(ctx, segmentHandler, outFilePattern, completionSignal)
 	}()
 
 	retryCount := 0
-	for retryCount < 5 {
+	for {
+		streamExists, err := ms.MediaMTXClient.StreamExists(ms.MediaMTXHost, id, sourceType)
```

Comment:
@j0sh not sure how you feel about this code having knowledge of mediamtx itself, maybe it could just be an interface instead `StreamChecker` or something with the `StreamExists` function.

Reply:
An interface is probably cleaner but I wouldn't spend too much time on it since the intent is to bring media handling in-house anyway and the rest of this code is becoming very livepeer-specific anyway with the logging stuff.

This bit is always going to be somewhat racy, eg it's possible the stream will disconnect between the StreamExists call and the reconnect. Do you know if the LPMS return is different if the stream exists vs not, or if the termination was clean vs not? If so, maybe it's worth just reconnecting on a non-nil error and checking the return rather than doing a separate stream exists check.

At some point we may also want to consider a "push" approach, eg having the runOnReady invoke ffmpeg to stream RTMP into go-livepeer, rather than having go-livepeer pull here. This might be a stepping stone to bringing more of the media handling in-house.